### ISSUE TITLE
notify when db is ready

### ISSUE THREAD

**Issue by nzfarmer1** [node-red/node-red-nodes#90]
Hi

I have tried using your level db from another module.

I can use this,       this.levelConfig = RED.nodes.getNode(n.level);

Then access this.levelConfig.db.get/put

However, there is a latency issue in that I am not aware when the db has been initialised.

Suggest emitting even here:

```
   lvldb(this.dbname, function(err, db) {
        if (err) { node.error(err); }
        node.db = db;
        node.emit("connect");  // Flag to other modules when db is ready!
    });
```

From my node I can then use the following during initialisation:

```
    this.levelConfig.on('connect', function() { ... });
```

Welcome your thoughts.

**Comment by dceejay** [comment]
as node.db only gets assigned within that same function can you not just check if it exists ? (I know it's not the node.js event style but... until we have a better plan)

**Comment by nzfarmer1** [comment]
thanks for your comment. do you have an example of how this would work? essentially i want to be able to make a db request during our node's init (at least prior to having this.on('input',...) invoked.

**Comment by dceejay** [comment]
maybe something like...

            var i = 0;
            var chk = function() {
                if (!node.levelConfig.db) {
                    i += 1;
                    if (i >= 20) {
                        node.error("Database didn't connect"); // report some lack of database error after 10 secs
                    }
                    setTimeout( function() {
                        chk();
                    }, 500); // check every 0.5 secs
                }
                else {
                    // do something when we see database
                }
            }
            chk();

**Comment by nzfarmer1** [comment]
yes, this would work.  to be honest i liked my suggestion better ...
thanks for the example.

**Comment by knolleary** [comment]
Yes, an event based solution is much cleaner than having to poll. But at
least you can poll today until we get to doing this properly.

**Comment by dceejay** [comment]
actually that code need to break out of the loop once it gets the timeout error... :-)

**Comment by nzfarmer1** [comment]
This adds backwards and hopefully forwards compatibility :) 

        int i =0;
        this.dbChk = function() { // node <=> this
            if (!node.levelConfig.db) {
                i += 1;
                if (i >= 20) {
                    node.error("Database didn't connect"); // report some lack of database error after 10 secs
                    return;
                }
                setTimeout( function() {
                    node.dbChk();
                }, 500); // check every 0.5 secs
            }
            else {
                node.levelConfig.emit('connect'); // Use existing handler.
            }
        }
        
        events.on("nodes-started",function(){
            node.dbChk();
        });
